internal/v4/api.go

```diff
@@ -31,25 +42,105 @@ type ReqHandler struct {
 }
 
 func New(pool *charmstore.Pool, config charmstore.ServerParams) Handler {
-	h := Handler{
+	return Handler{
 		Handler: v5.New(pool, config),
 	}
-	// TODO Set h.ResolveURL here.
-	return h
+}
+
+func (h Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	// When requests in this handler use router.RelativeURL, we want
+	// the "absolute path" there to be interpreted relative to the
+	// root of this handler, not the absolute root of the web server,
+	// which may be abitrarily many levels up.
+	req.RequestURI = req.URL.Path
+
+	rh, err := h.NewReqHandler()
+	if err != nil {
+		router.WriteError(w, err)
+		return
+	}
+	defer rh.Close()
+	rh.ServeHTTP(w, req)
 }
 
 func NewAPIHandler(pool *charmstore.Pool, config charmstore.ServerParams) charmstore.HTTPCloseHandler {
 	return New(pool, config)
 }
 
+// NewReqHandler fetchs a new instance of ReqHandler
```

Comment:
It doesn't really. It gets a store from h.Pool and then plugs it into a router from h.reqHandlerPool